apps/cowswap-frontend/src/modules/ordersTable/hooks/useOrdersHydrationState.ts

```diff
@@ -0,0 +1,43 @@
+import { useMemo } from 'react'
+
+import type { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+import { useSelector } from 'react-redux'
+
+import type { AppState } from 'legacy/state'
+import type { Order } from 'legacy/state/orders/actions'
+import { ContractDeploymentBlocks } from 'legacy/state/orders/consts'
+import type { OrdersStateNetwork } from 'legacy/state/orders/reducer'
+
+interface UseOrdersHydrationStateParams {
+  chainId: SupportedChainId | undefined
+  orders: Order[]
+}
+
+export function useOrdersHydrationState({ chainId, orders }: UseOrdersHydrationStateParams): boolean {
+  const networkState = useSelector<AppState, OrdersStateNetwork | undefined>((state) => {
+    if (chainId === undefined) {
+      return undefined
+    }
+
+    return state.orders?.[chainId as SupportedChainId]
+  })
+
+  const ordersLength = orders.length
+
+  return useMemo(() => {
+    if (chainId === undefined) {
+      return false
+    }
+
+    if (ordersLength > 0) {
+      return true
+    }
+
+    const chainKey = chainId as SupportedChainId
+    const lastCheckedBlock = networkState?.lastCheckedBlock
```

Comment:
`lastCheckedBlock` value is set in `PendingOrdersUpdater`, I don't understand why we should depend on it for the "NoOrders" image displaying

Reply:
That's a wrong statement. We don't need "account/network" to be fetched to display the animated image